sdk/go/pulumi-language-go/main.go

```diff
@@ -337,33 +376,50 @@ func (host *goLanguageHost) Run(ctx context.Context, req *pulumirpc.RunRequest)
 		return nil, errors.Wrap(err, "failed to prepare environment")
 	}
 
-	cmd, err := findProgram(host.binary)
-	if err != nil {
-		return nil, err
+	if os.Getenv("PULUMI_GO_USE_RUN") != "" {
```

Comment:
I was thinking if binary: is specified and PULUMI_GO_USE_RUN is specified, it will use run. But perhaps that's more intuitive! Good as-is I guess.

Reply:
this should be a bug. I just understood what you meant.